test/parallel/test-fs-mkdtemp-prefix-check.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const fs = require('fs');
+
+const expectedError = /^TypeError: filename prefix is required$/;
+const prefixValues = [undefined, null, 0, true, false, 1];
+
+function fail(value) {
+  assert.throws(
+    () => fs.mkdtempSync(value, {}),
+    expectedError
+  );
+}
+
+function failAsync(value) {
+  assert.throws(
+    () => fs.mkdtemp(value, () => {
+      common.fail('Async callback should not be called');
```

Comment:
`common.mustNotCall()` was added to prevent calling `common.fail()` in an otherwise empty function.